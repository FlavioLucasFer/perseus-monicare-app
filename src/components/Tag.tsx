import React from 'react';
import { Text, TextProps } from 'react-native';

interface Props extends TextProps {
	text: string,
	color: string,
};

const Tag: React.FC<Props> = props => {
	const {
		text,
		color,
		style,
	} = props;

	return (
		<Text 
			style={[{ 
					color: color, 
					backgroundColor: `${color}1A`, 
					padding: 2,
					paddingHorizontal: 7,
					borderColor: color,
					borderWidth: 1,
					borderRadius: 25,
					fontSize: 11,
				}, 
				style,
			]}
		>
			{text}
		</Text>
	);
};

export default Tag;
