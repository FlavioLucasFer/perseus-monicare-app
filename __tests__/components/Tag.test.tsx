import React from 'react';
import { shallow } from 'enzyme';

import Tag from 'components/Tag';

describe('Tag component', () => {
	it('Should render properly', () => {
		const wrapper = shallow(
			<Tag text='teste'
			  color='#FFFFFF'	/>
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});
});
