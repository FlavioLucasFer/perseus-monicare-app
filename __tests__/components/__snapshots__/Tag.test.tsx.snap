// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component Should render properly 1`] = `
"<Text style={{...}}>
  teste
</Text>"
`;
